### ISSUE TITLE
Mask to patch alignment not correct with register tokens?

### ISSUE THREAD

**Issue by massimilianoviola** [HKU-MMLab/OmniPart#12]
Hello @yhyang-myron! First of all, amazing work! I think I noticed a possible inconsistency in the way mask group embeddings are aligned with the DINO tokens. The model works great in practice, but if this is true, this is something that might be overlooked in future research, so please correct me if I am wrong or confirm the suspicion.

In both `forward()` ([L186](https://github.com/HKU-MMLab/OmniPart/blob/ceb1a4ff2d279a57a673d0f65954735c6bf8231c/modules/bbox_gen/models/autogressive_bbox_gen.py#L186C9-L186C57)) and `generate()` ([L245](https://github.com/HKU-MMLab/OmniPart/blob/ceb1a4ff2d279a57a673d0f65954735c6bf8231c/modules/bbox_gen/models/autogressive_bbox_gen.py#L245)) in `autogressive_bbox_gen.py` you do:

```python
masks_emb = rearrange(masks_emb, 'b c h w -> b (h w) c') # B x Q x C
group_emb = torch.zeros((image_latents.shape[0], image_latents.shape[1], masks_emb.shape[2]), device=image_latents.device, dtype=image_latents.dtype)
group_emb[:, :masks_emb.shape[1], :] = masks_emb  #<-- problem here? Shouldn't this be padded to the left instead of right?
image_latents = torch.cat([image_latents, group_emb], dim=-1)
```

Because the backbone is DINOv2-reg, the output token sequence is ordered as:
- CLS token at index 0
- Then num_register_tokens (=4) register tokens
- Then regular patch tokens

So writing `masks_emb` into `group_emb` starting at index 0 means the first group embeddings are aligned to CLS and register tokens, and in general, all mask entries are shifted by 1 + num_register_tokens. Thoughts?

**Comment by yhyang-myron** [comment]
Thanks a lot for the careful read and for flagging this — your reasoning makes a lot of sense.

We really appreciate you pointing this out. We’ll double-check the intended alignment in our implementation and make sure to account for this properly in our subsequent exploration. Thanks again for the great catch!
